packages/ragbits-document-search/src/ragbits/document_search/_main.py

```diff
@@ -141,8 +141,26 @@ async def ingest(
         elements = await self.processing_strategy.process_documents(
             documents, self.document_processor_router, document_processor
         )
+        await self.remove_entries_with_same_sources(elements)
         await self.insert_elements(elements)
 
+    async def remove_entries_with_same_sources(self, elements: list[Element]) -> None:
+        """
+        Remove entries from the vector store whose source id is present in the elements' metadata.
+
+        Args:
+            elements: List of elements whose source ids will be checked and removed from the vector store if present.
+        """
+        unique_source_ids = {element.document_meta.source.id for element in elements}
+
+        ids_to_delete = []
+        for entry in await self.vector_store.list():
+            if entry.metadata["document_meta"]["source"]["id"] in unique_source_ids:
```

Comment:
It will fail when entry doesn't have any of these keys. We cannot assume that only records coming from document search are in vector store.

Reply:
Corrected!